integration-tests/src/test_utils.rs

```diff
@@ -0,0 +1,634 @@
+use axelar_wasm_std::{nonempty, voting::PollID};
+use connection_router::state::{ChainName, CrossChainId, Message};
+use cosmwasm_std::{
+    coins, Addr, Attribute, Binary, BlockInfo, Deps, Env, Event, HexBinary, StdResult, Uint128,
+    Uint256, Uint64,
+};
+use cw_multi_test::{App, ContractWrapper, Executor};
+
+use k256::ecdsa;
+use multisig::key::PublicKey;
+use tofn::ecdsa::KeyPair;
+
+pub const AXL_DENOMINATION: &str = "uaxl";
+
+fn get_event_attribute<'a>(
+    events: &'a [Event],
+    event_type: &str,
+    attribute_key: &str,
+) -> Option<&'a Attribute> {
+    if let Some(event) = events.iter().find(|ev| ev.ty == event_type) {
+        return event
+            .attributes
+            .iter()
+            .find(|attr| attr.key == attribute_key);
+    }
+    None
+}
+
+pub fn verify_messages(app: &mut App, gateway_address: &Addr, msgs: &[Message]) -> (PollID, u64) {
+    let response = app
+        .execute_contract(
+            Addr::unchecked("relayer"),
+            gateway_address.clone(),
+            &gateway::msg::ExecuteMsg::VerifyMessages(msgs.to_vec()),
+            &[],
+        )
+        .unwrap();
+
+    let poll_id = get_event_attribute(&response.events, "wasm-messages_poll_started", "poll_id")
+        .map(|attr| serde_json::from_str(&attr.value).unwrap())
+        .expect("couldn't get poll_id");
+    let expiry = get_event_attribute(&response.events, "wasm-messages_poll_started", "expires_at")
+        .map(|attr| attr.value.as_str().parse().unwrap())
+        .expect("couldn't get poll expiry");
+    (poll_id, expiry)
+}
+
+pub fn route_messages(app: &mut App, gateway_address: &Addr, msgs: &[Message]) {
+    app.execute_contract(
+        Addr::unchecked("relayer"),
+        gateway_address.clone(),
+        &gateway::msg::ExecuteMsg::RouteMessages(msgs.to_vec()),
+        &[],
+    )
+    .unwrap();
+}
+
+pub fn vote_true_for_all(
+    app: &mut App,
+    workers: &Vec<Worker>,
+    voting_verifier_address: &Addr,
+    poll_id: PollID,
+    msgs: &Vec<Message>,
+) {
+    for worker in workers {
+        app.execute_contract(
+            worker.addr.clone(),
+            voting_verifier_address.clone(),
+            &voting_verifier::msg::ExecuteMsg::Vote {
+                poll_id,
+                votes: vec![true; msgs.len()],
+            },
+            &[],
+        )
+        .unwrap();
+    }
+}
+
+/// End the poll. Advances the current height to expiry if necessary
+pub fn end_poll(app: &mut App, voting_verifier_address: &Addr, poll_id: PollID, expiry: u64) {
+    if app.block_info().height < expiry {
+        advance_to_height(app, expiry);
+    }
```

Comment:
that's an oddly specific combination. I'd prefer to move that to it's own function and call
```
advance_at_least_to_height(app, expiry);
end_poll(...)
```